ocf_datapipes/transform/numpy_batch/datetime_features.py

```diff
@@ -0,0 +1,70 @@
+"""Datapipes to add Sun position to NumpyBatch"""
+
+import numpy as np
+from torch.utils.data import IterDataPipe, functional_datapipe
+
+from ocf_datapipes.batch import BatchKey
+
+
+def _get_date_time_in_pi(dt):
+    day_of_year = (dt - dt.astype("datetime64[Y]")).astype(int)
+    minute_of_day = (dt - dt.astype("datetime64[D]")).astype(int)
+
+    # converting into positions on sin-cos circle
+    time_in_pi = (2 * np.pi) * (minute_of_day / (24 * 3600))
+    date_in_pi = (2 * np.pi) * (day_of_year / (365 * 24 * 3600))
+
+    return date_in_pi, time_in_pi
+
+
+@functional_datapipe("add_trigonometric_date_time")
+class AddTrigonometricDateTimeIterDataPipe(IterDataPipe):
+    """Adds the trigonometric encodings of date of year, time of day to the NumpyBatch"""
+
+    def __init__(self, source_datapipe: IterDataPipe, modality_name: str):
+        """
+        Adds the sine and cosine of time to the NumpyBatch
+
+        Args:
+            source_datapipe: Datapipe of NumpyBatch
+            modality_name: Modality to add the time for
+        """
+        self.source_datapipe = source_datapipe
+        self.modality_name = modality_name
+        assert self.modality_name in [
+            "wind",
+        ], f"Trigonometric time not implemented for {self.modality_name}"
+
+    def __iter__(self):
+        for np_batch in self.source_datapipe:
+            time_utc = np_batch[BatchKey.wind_time_utc]
+
+            # Check if the input is batched
+            # time_utc could have shape (batch_size, n_times) or (n_times,)
+            assert len(time_utc.shape) in [1, 2]
+            is_batched = len(time_utc.shape) == 2
+
+            times = time_utc.astype("datetime64[s]")
+
+            if is_batched:
```

Comment:
Do we expect that the times will be an a numpy array here? If so would the broadcasting of the `_get_date_time_in_pi` just work for the 2 dimensional case as well? I _think_ it might so you may not need this special logic for the is batched case? Let me know if I am mistaken or I have missed something!